modules/c++/cli/source/ArgumentParser.cpp

```diff
@@ -437,24 +472,34 @@ cli::Results* cli::ArgumentParser::parse(const std::vector<std::string>& args)
                 }
                 else
                 {
-                    throw except::Exception(Ctxt(FmtX("Invalid option: [%s]",
-                                                      argStr.c_str())));
+                    if (mIgnoreUnknownArguments)
+                    {
+                        mIgnoreUnknownOStream
+                                << "Unknown arg: " << argStr.c_str()
+                                << std::endl;
+                        break;
```

Comment:
This break will cause all arguments after this one to be ignored. Changing it to `continue` will allow it to process the remaining arguments.

There are also some more complicated cases to consider, like if a user tries to pass an argument which does not exist with a following value (something like `--filename C:/Data/File.txt`).